app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package org.oppia.app.player.audio
+
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import javax.inject.Inject
+
+private const val TAG_DIALOG = "LANGUAGE_DIALOG"
+
+/** [ViewModel] for user app usage history. */
+@FragmentScope
+class AudioViewModel @Inject constructor(
+) : ViewModel() {
+
+  private var currentLanguageCode: String = "en"
+
+  val audioLanguageCode: String? by lazy {
+    getLanguageCode()
+  }
+
+  fun setAudioLanguageCode(languageCode: String) {
+    currentLanguageCode = languageCode
+  }
+
+  private fun getLanguageCode(): String {
+    return currentLanguageCode
+  }
+
+  fun languageSelectionClicked(fragmentManager: FragmentManager) {
+    val previousFragment = fragmentManager.findFragmentByTag(TAG_DIALOG)
+    if (previousFragment != null) {
+      fragmentManager.beginTransaction().remove(previousFragment).commitNow()
```

Comment:
Ah. I really think it would be nicer to use dismiss(), but I just looked through the AndroidX code and it appears dismiss() is not synchronous (it uses commit() internally), so this is unfortunately necessary.

Reply:
That means this is correct right?